media/js/src/simulations/simulation4/whatAreNonlinearRegs.jsx

```diff
@@ -2,20 +2,32 @@ import React from 'react';
 import { PromptBlock } from '../../PromptBlock';
 import PropTypes from 'prop-types';
 import { Katex } from '../../utils/katexComponent';
+import dataset from './polynomial.json';
+
 
 export const WhatAreNonLinearRegressions = ({
     setshowRegLine, setshowDatasets, showRegLine, showDatasets,
     setMysteryRegLine, mysteryRegLine}) => {
 
     const DATASET_NAMES = ['Linear','Quadratic','Cubic'];
 
-    const formulas = [
-        '\\hat{y} = \\hat{\\beta}_0 + \\hat{\\beta}_1 X + u',
-        // eslint-disable-next-line max-len
-        '\\hat{y} = \\hat{\\beta}_0 + \\hat{\\beta}_1 X + \\hat{\\beta}_2 X^2 + u',
-        // eslint-disable-next-line max-len
-        '\\hat{y} = \\hat{\\beta}_0 + \\hat{\\beta}_1 X + \\hat{\\beta}_2 X^2 + \\hat{\\beta}_3 X^3 + u',
-    ];
+    const getFormula = (key) => {
+        const data = dataset[key];
+
+        if (key === 'linear') {
+            return `\\hat{y} = ${data.intercept.toFixed(2)} +
+            ${data.slope.toFixed(2)}X`;
+        } else if (key === 'quadratic') {
+            return `\\hat{y} = ${data.intercept.toFixed(2)} +
+            ${data.slope.toFixed(2)}X + ${data.slope2.toFixed(2)}X^2`;
+        } else if (key === 'cubic') {
+            return `\\hat{y} = ${data.intercept.toFixed(2)} +
+            ${data.slope.toFixed(2)}X + ${data.slope2.toFixed(2)}X^2 +
+            ${data.slope3.toFixed(2)}X^3`;
```

Comment:
If you'd like I can round those values to the nearest hundredth so we don't need to run `toFixed()` on everything

Reply:
Yeah we can do that later on if this looks okay to Seyhan